### PR TITLE
added support to network address range for multiple apps on same account

### PR DESCRIPTION
**Pull request by msaaditoluna**

### CODE REVIEW

**Review by Eli-Meitner — APPROVED** [review]

**Review by srinivas-toluna — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -70,3 +74,6 @@ variable "default_network_acl_egress" {
     description = "List of ACL egress rules"
     default = [{ "rule_no": 100, "action": "allow", "cidr_block": "0.0.0.0/0", "from_port": 0, "protocol": "-1",  "to_port": 0 } ]
 }
+variable "lookup_ssm_param" {
```
Comment (raanan245) [inline variables.tf:77]: I dont like this variable name - its says nothing regarding its purpose (especially when the default is true)
In general I prefer a solution without explicitly enabling a flag.

File: main.tf [hunk 2]
```diff
@@ -3,7 +3,7 @@ locals {
   aws_azs = slice(data.aws_availability_zones.available.names[*], 0, var.number_of_azs)
 
   # calculate VPC and subnets CIDRs
-  vpc_cidr         = cidrsubnet(data.aws_ssm_parameter.network_range.value, 5, var.env_index)
+  vpc_cidr         = "${ var.lookup_ssm_param == false ? cidrsubnet(data.aws_ssm_parameter.network_range.value, 5, var.env_index) : cidrsubnet(data.aws_ssm_parameter.network_range_per_app[0].value, 5, var.env_index) }"
```
Comment (raanan245) [inline main.tf:6]: is there no way to have an empty data source ? (without using the lookup var)
To make the code more clean and readable I would also have this condition as a separate line for setting the vpc_cidr and then do the cidrsubset function

File: outputs.tf [hunk 3]
```diff
@@ -37,4 +37,6 @@ output "default_dhcp_options" {
 output "aws_region" {
     value = data.aws_region.current.name
 }
-
+# output "network_address_range" {
+#   value = "${ var.lookup_ssm_param == false ? cidrsubnet(data.aws_ssm_parameter.network_range.value, 5, var.env_index) : cidrsubnet(data.aws_ssm_parameter.network_range_per_app[0].value, 5, var.env_index) }"
```
Comment (raanan245) [inline outputs.tf:41]: please have this value conditional calculation done only once in the local block and not duplicate this code.

**Comment by raanan245** [comment]
I know you already merged it, but I think more changes should be done, and would re-open this PR.
In addition to the inline comments I've added on my review, I think you're also missing the usage of the app_name var as a unique ID for the resources created by the VPC module. Including its sub-modules.

**Pull request merged**